Only zero out reader buffer sizes if set.

Shelnutt2 opened the pull request:
This fixes a segfault from when buffer_var_size_ is not set but it is dereferenced. A unit test is also added for the sequence to create this segfault.

Fixes #606

tdenniston commented:
@Shelnutt2 This looks good, please rebase onto `dev` and we'll merge this one before the other open PR.

Shelnutt2 commented:
@tdenniston I've rebased on dev, should be good now.

For the rebasing, mind if I ask why you want the branch rebased against the latest dev? If there are no merge conflicts, then when travis ci/appveyor runs the pull request build, it builds and tests the resulting merge commit, so it should be validating what the code will be after the merge. I don't mind rebasing the branches so its always built ontop of the latest dev, just curious to the rationale and if I'm missing another reason for keeping it rebased.

jakebolewski commented:
I think you have to force push the rebase to this PR branch

In general that is true, but I don't know if Travis will re build the PR if dev changes after the PR was issued (it may)

Shelnutt2 commented:
I've rebased it again, looks like there were two more commits between when I rebased earlier and now.

For travis ci it appears it will not currently re-run the tests when the dev branch has changes, there are two issues for this features https://github.com/travis-ci/travis-ci/issues/1620 and https://github.com/travis-ci/travis-ci/issues/9197 . In that case rebase to always rerun the unit tests makes sense.
